rusk-wallet/src/gql.rs

```diff
@@ -58,14 +61,50 @@ struct BlockResponse {
     pub block: Option<Block>,
 }
 
+// See `PhoenixTransactionEvent` for the reason for this struct
+// and allowing dead code here.
 #[serde_as]
-#[derive(Deserialize, Debug)]
-pub struct BlockData {
-    #[serde_as(as = "DisplayFromStr")]
-    pub gas_spent: u64,
-    pub sender: String,
+#[derive(Deserialize, Debug, Clone)]
+#[allow(dead_code)]
+struct Note {
+    stealth_address: StealthAddress,
+}
+
+// This struct is used instead of the one in `dusk_core::transfer`
+// because of the order-dependent deserialization bug in
+// the `dusk_core::transfer::phoenix::Note` https://github.com/dusk-network/phoenix/issues/274.
+// Dead code is allowed to avoid catch-alls, so that the case in which an
+// unexpected event is received, an appropriate error will be thrown.
+#[serde_as]
+#[derive(serde::Deserialize, Debug)]
+#[allow(dead_code)]
+pub struct PhoenixTransactionEvent {
+    /// Notes produced during the transaction.
+    notes: Vec<Note>,
```

Comment:
These are not notes but note-addresses
```suggestion
    note_addressess: Vec<NoteAddress>,
```